### ISSUE TITLE
[Question] heat mechanics

### ISSUE THREAD

**Issue by Briaireous** [tsunamayo/Starship-EVO#5035]
Hello tsuna, just got two questions in regards to the current Heat mechanic.

How does the game decide on how much Heat dissipation a block provides to a build?
Reason being is that I'm trying to min/max the base stats on a fighter I'm building and the value of Heat dissipation varied by many magnitudes depending on where I place the blocks. I could place down a 1m block and have another thousand Heat dissipation per second, or I could have just and additional 10 head dissipation per second. I have even managed to decrease Heat dissipation of the build by building a 1m block stretched to 2mx1mx1m.

Since you had once put out a Heat system revamp and then reverted it soon after. Are there any future plans for another change Heat system?

**Comment by Dwarf-LordPangolin** [comment]
Am not Tsuna, but:

Based on my own experience trying to min-max fighters, the bounding box of the vehicle is the primary determinant of heat dissipation when calculating surface area. As you may know, surface area dissipation got significantly buffed some time ago, because prior to that point fighters made so much heat they were blowing themselves up just from existing. Buffing surface area dissipation made fighters viable without affecting the balance for larger ships, because the bigger the ship, the faster volume increases compared to surface area. This means that if you increase the ship's surface area, even by a relatively small amount, it can dissipate a lot more heat.

Now, I don't know how the game precisely calculates surface area, but my own experiments suggest that the ship's bounds are the main factor for working it out. For example, I'm working on a 125m long ship ATM, and I just cut its heat from 79% to 73% by sticking an 8m long rod on the nose -- thereby extending the bounding box by 8 meters. Just adding a .5x.5x8m rod took its dissipation from 159,556 to 173,488. But if I put that same rod flat on the hull, inside the current bounding box, it does almost nothing, because then the only effect it has on heat dissipation is from mass -- which did not get the same buff that surface area did, because big ships weren't exploding themselves.

BTW, this is the same reason that child entities currently give huge buffs to heat dissipation -- the parent entity inherits the dissipation of the child entity, but the child entity's surface area is calculated independently of the parent's volume, so your landing gear provide the same dissipation they would if they were extending the ship's bounding box.

[When last I heard](https://github.com/tsunamayo/Starship-EVO/issues/3444), Tsuna was aware of that issue, and it is on his to-do list for when he reworks heat mechanics. I think he's focusing more on adding game features ATM than balance.

**Comment by Briaireous** [comment]
> Am not Tsuna, but:
> 
> Based on my own experience trying to min-max fighters, the bounding box of the vehicle is the primary determinant of heat dissipation when calculating surface area. As you may know, surface area dissipation got significantly buffed some time ago, because prior to that point fighters made so much heat they were blowing themselves up just from existing. Buffing surface area dissipation made fighters viable without affecting the balance for larger ships, because the bigger the ship, the faster volume increases compared to surface area. This means that if you increase the ship's surface area, even by a relatively small amount, it can dissipate a lot more heat.
> 
> Now, I don't know how the game precisely calculates surface area, but my own experiments suggest that the ship's bounds are the main factor for working it out. For example, I'm working on a 125m long ship ATM, and I just cut its heat from 79% to 73% by sticking an 8m long rod on the nose -- thereby extending the bounding box by 8 meters. Just adding a .5x.5x8m rod took its dissipation from 159,556 to 173,488. But if I put that same rod flat on the hull, inside the current bounding box, it does almost nothing, because then the only effect it has on heat dissipation is from mass -- which did not get the same buff that surface area did, because big ships weren't exploding themselves.
> 
> BTW, this is the same reason that child entities currently give huge buffs to heat dissipation -- the parent entity inherits the dissipation of the child entity, but the child entity's surface area is calculated independently of the parent's volume, so your landing gear provide the same dissipation they would if they were extending the ship's bounding box.
> 
> [When last I heard](https://github.com/tsunamayo/Starship-EVO/issues/3444), Tsuna was aware of that issue, and it is on his to-do list for when he reworks heat mechanics. I think he's focusing more on adding game features ATM than balance.

yeah, heard tsuna had covid and some family time. Also heard he is working on planets
